src/app/modules/shared/components/breadcrumb/breadcrumb.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable, EventEmitter } from '@angular/core'
+import { Router, ActivatedRouteSnapshot, Event, NavigationEnd, UrlSegment } from '@angular/router'
+import { Breadcrumb } from './breadcrumb.model'
+
+@Injectable()
+export class BreadcrumbService {
+  breadcrumbChanged = new EventEmitter<Breadcrumb[]>(false);
+
+  private breadcrumbs: Breadcrumb[] = [];
+
+  constructor(private router: Router) {
```

Comment:
Since services are not components and don't have lifecycle hooks, doing subscriptions inside of a service constructor can be problematic as it might not always work or lead to unexpected issues. Instead, we should allow the component calling this service to subscribe to route events